### PR TITLE
refactor: Move cryptobox hooks into @wireapp/core

### PR DESCRIPTION
**Pull request by atomrc**
Will let the core handle session creation and prekey generation hooks.
needs https://github.com/wireapp/wire-web-packages/pull/4535

### CODE REVIEW

**Review by aweiss-dev — APPROVED** [review]

File: src/script/cryptography/CryptographyRepository.ts [hunk 1]
```diff
@@ -47,21 +43,9 @@ export interface ClientKeys {
...
   logger: Logger;
 
-  static get CONFIG() {
-    return {
-      UNKNOWN_DECRYPTION_ERROR_CODE: 999,
-    };
-  }
-
-  static get REMOTE_ENCRYPTION_FAILURE(): string {
-    return '💣';
```
Comment (PatrykBuniX) [inline src/script/cryptography/CryptographyRepository.ts:62]: wow 💣

File: src/script/cryptography/CryptographyRepository.ts [hunk 2]
```diff
@@ -175,10 +142,11 @@ export class CryptographyRepository {
         );
         const sessionId = this.core.service!.cryptography.constructSessionId({domain, id: userId}, clientId);
         const preKeyArray = base64ToArray(preKey.key);
-        return await this.cryptobox!.session_from_prekey(sessionId, preKeyArray.buffer);
+        return await this.cryptobox.session_from_prekey(sessionId, preKeyArray.buffer);
```
Comment (przemvs) [inline src/script/cryptography/CryptographyRepository.ts:145]: await there is not needed :)

Comment (atomrc) [reply src/script/cryptography/CryptographyRepository.ts:145]: true that :)

File: src/script/main/app.ts [hunk 3]
```diff
@@ -152,8 +152,8 @@ class App {
     storage: StorageService;
   };
   repository: ViewModelRepositories = {} as ViewModelRepositories;
-  debug: DebugUtil;
-  util: {debug: DebugUtil};
+  debug?: DebugUtil;
+  util?: {debug: DebugUtil};
```
Comment (przemvs) [inline src/script/main/app.ts:156]: Shouldn't be moved after required types?

**Review by przemvs — APPROVED** [review]
Few things, rest LGTM :)

File: src/script/cryptography/CryptographyRepository.ts (hunk 2 above)
Comment (atomrc) [reply src/script/cryptography/CryptographyRepository.ts:145]: actually I do need it 
image

File: src/script/main/app.ts (hunk 3 above)
Comment (atomrc) [reply src/script/main/app.ts:156]: It's an object, so the order is not important

**Review by PatrykBuniX — APPROVED** [review]

**Pull request merged**